### ISSUE TITLE
I'd love to use this crate, but I can't because it brings in a `std` dependency and I have to be `#[no_std]`

### ISSUE THREAD

**Issue by alexschrod** [mlund/mos-hardware#1]
The culprit is the `either` crate:
```
error[E0463]: can't find crate for `std`
  --> /home/mos/.cargo/registry/src/github.com-1ecc6299db9ec823/either-1.8.0/src/lib.rs:19:1
   |
19 | extern crate std;
   | ^^^^^^^^^^^^^^^^^ can't find crate
   |
   = note: the `mos-cx16-none` target may not support the standard library
   = help: consider building the standard library from source with `cargo build -Zbuild-std`
```
which is brought in as soon as I depend on `mos-hardware`:
```
$ cargo tree -i either -e features
either v1.8.0
├── either feature "default"
│   └── which v4.3.0
│       └── bindgen v0.60.1
│           ├── bindgen feature "clap"
│           │   └── bindgen feature "default"
│           │       [build-dependencies]
│           │       └── mos-hardware v0.1.9
│           │           └── mos-hardware feature "default"
│           │               └── my_crate v0.0.0 (/workspaces/my_crate)
│           │                   └── my_crate feature "default" (command-line)
│           ├── bindgen feature "default" (*)
│           ├── bindgen feature "env_logger"
│           │   └── bindgen feature "logging"
│           │       └── bindgen feature "default" (*)
│           ├── bindgen feature "log"
│           │   └── bindgen feature "logging" (*)
│           ├── bindgen feature "logging" (*)
│           ├── bindgen feature "runtime"
│           │   └── bindgen feature "default" (*)
│           ├── bindgen feature "which"
│           │   └── bindgen feature "which-rustfmt"
│           │       └── bindgen feature "default" (*)
│           └── bindgen feature "which-rustfmt" (*)
└── either feature "use_std"
    └── either feature "default" (*)

either v1.8.0
└── itertools v0.10.5
    └── mos-hardware v0.1.9 (*)
```

**Comment by mlund** [comment]
Hmm, this crate is `no_std`, so that shouldn't a problem. Do you use [rust-mos](https://llvm-mos.org/wiki/Rust) to build? It will not work with a "normal" rust installation.

**Comment by alexschrod** [comment]
Yeah, I'm using rust-mos. Hence the
```
note: the `mos-cx16-none` target may not support the standard library
```

**Comment by mlund** [comment]
OK, let me have a look. I suppose you're using `mos-hardware` as a dependency? I have never tried that. Could you checkout the `mos-hardware` repository and test to see if it compiles with `cargo build`?

**Comment by alexschrod** [comment]
Yeah, if I check out this repo and try to build it, it builds fine. But when I put this in my project's Cargo.toml:
```toml
[dependencies]
mos-hardware = "0.1"
```
and try to build it, I get the error from `either` as mentioned in my OP.

For the time being I've worked around this by copying the relevant code into my own project, but it'd feel more "right" to use this code as a dependency instead of copied in.

**Comment by alexschrod** [comment]
I figured it out! My project is in a workspace with no root crate, and apparently, that configuration will use Cargo's `resolver = "1"` by default, which explains why the build dependency on `either` had an effect on the regular dependency. I added
```toml
[workspace]
resolver = "2"
```
under my `[workspace]` table and it behaved like normal.

That's really unfortunate default behavior tbh... but at least I figured it out.
